src/compiler/scala/tools/nsc/typechecker/Unapplies.scala

```diff
@@ -154,6 +156,10 @@ trait Unapplies extends ast.TreeDSL {
     val mods =
       if (applyShouldInheritAccess(inheritedMods))
         (caseMods | (inheritedMods.flags & PRIVATE)).copy(privateWithin = inheritedMods.privateWithin)
+          .tap { mods =>
+            if (currentRun.isScala3 && mods != caseMods)
+              runReporting.warning(cdef.pos, "constructor modifiers are assumed by synthetic `apply` method", WarningCategory.Scala3Migration, cdef.symbol)
```

Comment:
**question:** what's the solution to this warning? I'm a bit confused by the message...

Reply:
discord is pretty good for questions. There is a ticket to improve this warning.

Reply:
@som-snytt https://github.com/scala/bug/issues/12883 , or do you have some other ticket in mind?

Reply:
I didn't want to create needless cross-reference noise... Probably this sounds like a "cross" reference.